dts/bindings/clock/st,stm32-clock-mco.yaml

```
@@ -0,0 +1,43 @@
+#
+# Copyright (c) 2024, Joakim Andersson
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+compatible: "st,stm32-clock-mco"
+
+description: |
+        STM32 Microcontroller clock output (MCO)
+        Configure the microcontroller clock output pin and additional properties,
+        e.g. clock source and prescaler.
+        Example:
+                &mco {
```

Nit: since the node is always named `mco1`
```suggestion
                &mco1 {
```